repos/system_upgrade/common/actors/checkfirstpartitionoffset/libraries/check_first_partition_offset.py

```diff
@@ -0,0 +1,64 @@
+from leapp import reporting
+from leapp.libraries.common import grub
+from leapp.libraries.stdlib import api, CalledProcessError, run
+
+SAFE_OFFSET_kiB = 1024  # 1MiB
+
+
+def is_first_partition_offset_ok(device):
+    try:
+        partition_table = run(['parted', device, 'unit', 'kiB', 'print'], split=True)['stdout']
+    except CalledProcessError:
+        # Unlikely - if the disk has no partition table, `parted` terminates with 0 (no err). Parted exits with an err
+        # when the device does not exists, or if it is too small to contain a partition table.
+        api.current_logger().error('Failed to run `parted` to obtain partition table of the device.')
+        return True  # There is no first partition, so the predicate holds vacuously
```

Comment:
Two things herE:
* Add more information to the log,  For instance, the string representation of the exception and the name of the device.  If we encounter a case where we think parted should have found a partition table but it did not, we will want more information about what happened.
* Don't think we should `return True` here.  You list two cases that `parted` could exit with an error:
  * In the case the device does not exist, the caller should make the decision of what to do (It has to think about whether the devices it is passing in could be invalid or not and act appropriately)  So this case should raise an exception.
  * In the case the device is too small to contain a partition table, grub might also be too big to fit on the device, so I think this case should do one of the following:
    * `return False` (if the function is deciding whether there is space to place grub)
    * do some additional checking to see if there is room for grub before choosing whether to return True or False
    * or raise an exception for the caller to decide.

Reply:
I've decided to raise exception that will eventually lead to leapp terminating with `StopActorExecutionError`. The reason is that the root cause of the problem -- the `get_grub_devices` function -- is also used later to update the grub core. If the function is (unintentionally) lying, then it is better to halt the upgrade and be aware of the problem, rather than attempting to install a boot loader to a device that, e.g., does not exist.